### PR TITLE
Initial ElectroNP flowsheet

### PR DESCRIPTION
**Pull request by luohezhiming**
## Fixes/Resolves:

(replace this with the issue # fixed or resolved, if no issue exists then a brief statement of what this PR does)

## Summary/Motivation:
Initial implementation of electroNP flowsheet with anaerobic digester followed by electroNP unit.

## Changes proposed in this PR:
- Add electroNP flowsheet
- Add tests
- Add scaling factor for electroNP_ZO
- Modify tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        # if "conc_mass_comp[X_PAO]" in var.name:
+        #     iscale.set_scaling_factor(var, 1e3)
+        # if "conc_mass_comp[X_AUT]" in var.name:
+        #     iscale.set_scaling_factor(var, 1e3)
+        # if "conc_mass_comp[X_MeOH]" in var.name:
+        #     iscale.set_scaling_factor(var, 1e3)
+        # if "conc_mass_comp[X_MeP]" in var.name:
+        #     iscale.set_scaling_factor(var, 1e3)
+        # if "conc_mass_comp[X_TSS]" in var.name:
+        #     iscale.set_scaling_factor(var, 1e3)
```
Comment (MarcusHolly) [inline watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py]: Why are these commented out?

**Review by MarcusHolly — APPROVED** [review]
With the exception of one small comment, this LGTM.

**Review by agarciadiego — APPROVED** [review]

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    ModifiedASM2dParameterBlock,
+)
+from watertap.unit_models.translators.translator_adm1_asm2d import Translator_ADM1_ASM2D
+from watertap.unit_models.electroNP_ZO import ElectroNPZO
+from idaes.core.util.tables import (
+    create_stream_table_dataframe,
+    stream_table_dataframe_to_string,
+)
+from watertap.costing import WaterTAPCosting
+from watertap.core.util.model_diagnostics.infeasible import *
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py]:
```suggestion
```
You can and should probably remove any debugging tools that you used and kept in the flowsheet

File: watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py (hunk 1 above)
Comment (adam-a-a) [reply watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet.py]: delete or set scaling factors to 1 and revise later

**Review by adam-a-a — APPROVED** [review]
Great job on this!

**Pull request merged**